libs/tx-blockchain/src/fetch.ts

```diff
@@ -102,66 +88,20 @@ async function readAndMergeTransfers(account: string, to: boolean, fromBlock: nu
   return mergeTransactions(history, txs);
 }
 
-//async function  purchaseToTransaction(account: string, ethersLog: Log, contract: Contract): Promise<Transaction> {
-//   const res = contract.interface.parseLog(ethersLog);
-//   const {purchaser, amount, balance, timestamp} = res.values;
-//   if (account != purchaser)
-//     return null;
-//   return {
-//     txHash: ethersLog.transactionHash,
-//     date: new Date(timestamp.toNumber() * 1000),
-//     change: amount.toNumber(),
-//     logEntry: `Purchase: ${amount.toNumber()}`,
-//     balance: balance.toNumber()
-//   }
-// }
-
-//async function  readAndMergePurchases(account: string, fromBlock: number, contract: Contract, history: Transaction[]) {
-//   // construct filter to get tx either from or to
-//   let filter: any = contract.filters.Purchase();
-//   filter.fromBlock = fromBlock || 0;
-//   // Retrieve logs
-//   const txLogs = await contract.provider.getLogs(filter)
-//   // Convert logs to our transactions
-//   let txs: Transaction[] = [];
-//   for (let i = 0; i < txLogs.length; i++) {
-//     const tx = await purchaseToTransaction(account, txLogs[i], contract);
-//     if (tx)
-//       txs.push(tx);
-//   }
-//   // merge and remove duplicates for complete array
-//   return mergeTransactions(history, txs);
-// }
-
-export async function loadAndMergeHistory(address: string, fromBlock: number, contract: TheCoin, history: Transaction[]) {
+export async function loadAndMergeHistory(address: string, fromBlock: number, contract: TheCoin) {
   try {
-    //history = await readAndMergePurchases(address, fromBlock, contract, history);
+    let history: Transaction[] = [];
     history = await readAndMergeTransfers(address, true, fromBlock, contract, history);
     history = await readAndMergeTransfers(address, false, fromBlock, contract, history);
+    return history;
   }
   catch (err) {
-    console.error(err);
+    log.error(err);
```

Comment:
Question: is it better to catch or let any exceptions propagate.

Our exception handling is not consistent